### ISSUE TITLE
Type error involving overloaded function and float pow

### ISSUE THREAD

**Issue by wj32** [microsoft/pyright#2643]
I believe this issue only started occurring with pyright 1.1.191.

My understanding is that if `x` and `y` are `float`s then `x ** y` should be `float`. However, running pyright on the following code produces an error:

```
error: Argument of type "A" cannot be assigned to parameter "b" of type "B" in function "accept_b"
    "A" is incompatible with "B" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 infos 
```

```python
from typing import Union, overload

class A:
    pass

class B:
    pass

def accept_a(a: A):
    ...

def accept_b(b: B):
    ...

def accept_float(x: float):
    ...

@overload
def func(lhs: A, rhs: Union[A, B, float]) -> A:
    ...

@overload
def func(lhs: Union[A, B, float], rhs: A) -> A:
    ...

@overload
def func(lhs: Union[B, float], rhs: Union[B, float]) -> B:
    ...

def func(lhs: Union[A, B, float], rhs: Union[A, B, float]) -> Union[A, B]:
    ...

def test(x: float, y: float):
    accept_float(x)
    accept_float(x ** y)
    accept_b(func(B(), x))
    accept_b(func(B(), x ** y)) # Error here - pyright thinks func(B(), x ** y) has type A
```

**Comment by erictraut** [comment]
This is due to the following function declaration in the typeshed builtins.pyi:

```python
class float:
    ...
    @overload
    def __pow__(self, __x: int, __mod: None = ...) -> float: ...
    # positive x -> float; negative x -> complex
    # return type must be Any as `float | complex` causes too many false-positive errors
    @overload
    def __pow__(self, __x: float, __mod: None = ...) -> Any: ...
```

So the type of `x ** y` is `Any`. The challenge is that at runtime it sometimes returns `float` and sometimes returns `complex` depending on the sign of the value `y`.

For more information about the reasoning behind this, check out the following:
https://github.com/python/typeshed/issues/1406
https://github.com/python/typeshed/issues/4352

**Comment by erictraut** [comment]
This changed recently in typeshed, and pyright 1.1.191 included these recent changes. If you have concerns about the change in typeshed, please post an issue in the [typeshed github repo](https://github.com/python/typeshed).

**Comment by wj32** [comment]
Thanks @erictraut for the quick response! That makes sense.
